### PR TITLE
deSEC implement pagination

### PR DESCRIPTION
**Pull request by D3luxee**
This PR will implement pagination for the deSEC provider (#1177), this is needed when ever the api could return over 500 elements.

### CODE REVIEW

File: providers/desec/protocol.go [hunk 1]
```diff
@@ -71,37 +73,138 @@ func (c *desecProvider) authenticate() error {
 	}
 	return nil
 }
+func (c *desecProvider) initializeDomainIndex() error {
+	if c.domainIndexInitialized {
```
Comment (tlimoncelli) [inline providers/desec/protocol.go:77]: Suggestion: Protect domainIndex with a mutex.  It isn't needed now, but some day we'll call these concurrently.

File: providers/desec/protocol.go [hunk 2]
```diff
@@ -16,9 +18,9 @@ const apiBase = "https://desec.io/api/v1"
 
 // Api layer for desec
 type desecProvider struct {
-	domainIndex      map[string]uint32 //stores the minimum ttl of each domain. (key = domain and value = ttl)
-	nameserversNames []string
-	creds            struct {
+	domainIndex            map[string]uint32 //stores the minimum ttl of each domain. (key = domain and value = ttl)
+	domainIndexInitialized bool              // if the domainIndex was fully initialized once
```
Comment (tlimoncelli) [inline providers/desec/protocol.go:22]: Do we need `domainIndexInitalized`?  Could we simply [check if it is non-nil?](https://stackoverflow.com/questions/31339249/check-if-a-map-is-initialised-in-golang)

Comment (D3luxee) [reply providers/desec/protocol.go:22]: I've changed it.

File: providers/desec/protocol.go (hunk 1 above)
Comment (D3luxee) [reply providers/desec/protocol.go]: Done

**Review by tlimoncelli — APPROVED** [review]
Looks great!  I have no further comments.  Anything else or is it ready to merge?

**Comment by D3luxee** [comment]
It can be merged

**Comment by tlimoncelli** [comment]
Thanks for fixing this!  Big sites really appreciate it!

**Pull request merged**